include/avif/internal.h

```diff
@@ -282,7 +282,7 @@ avifBool avifSequenceHeaderParse(avifSequenceHeader * header, const avifROData *
 
 // A maximum image size to avoid out-of-memory errors or integer overflow in
 // (32-bit) int or unsigned int arithmetic operations.
-#define AVIF_MAX_IMAGE_SIZE (16384 * 16384)
```

Comment:
The current max image size of 16384 * 16384 (= 2^28) can be justified as follows:

> INT_MAX is 2^31 - 1. We want to allow our clients to compute the buffer size of four planes (RGB and alpha) of bit depth 10 or 12 using the 'int' type. So 4 * 2 * AVIF_MAX_IMAGE_SIZE < 2^31, which implies AVIF_MAX_IMAGE_SIZE < 2^28.

The current AVIF_MAX_IMAGE_SIZE value is already lenient because it allows 4 * 2 * width * height to be 2^31, which is greater than INT_MAX by 1. Therefore, we cannot increase AVIF_MAX_IMAGE_SIZE by four times.

The use of the 'int' type for width and height is common, so this integer overflow is a real security concern. If the size_t type is used instead, the current AVIF_MAX_IMAGE_SIZE value also helps avoid size_t overflow on 32-bit platforms.

Now, for libavif clients that know how to avoid integer overflows, we can make this an option in avifDecoder to allow a larger max image size on 64-bit platforms.

Note: Another consideration when choosing a max image size is to avoid memory allocation failures in avifAlloc(). avifAlloc() calls abort() rather than return NULL when memory allocation fails. It is important to avoid passing a huge 'size' argument to avifAlloc().